R/salmon.R

```diff
@@ -24,10 +24,18 @@ salmon_counts <- function(x, tx2gene = NULL) {
     all(colnames(tx2gene) == c("tx_name", "gene_id")),
     msg = "The tx2gene df object has incorrect column names."
   )
-
-  txi_salmon <- tximport::tximport(files = x, type = "salmon", tx2gene = tx2gene)
-  counts <- txi_salmon[["counts"]] |>
-    tibble::as_tibble(rownames = "rowname", .name_repair = make.names) |>
-    dplyr::rename(count = .data$X)
+  # check if gene level counts are provided or transcript level
+  if (grepl("genes.sf", basename(x), fixed = TRUE)) {
+    counts <- read.table(x, sep="\t", as.is=TRUE, header=TRUE) |>
```

Comment:
```suggestion
    counts <- readr::read_tsv(x, col_types = readr::cols(.default = "c", NumReads = "d")) |>
```

Reply:
@pdiakumis - what's `c` and `d` in `cols`?

Reply:
Ah, abbreviations for character and double - see https://readr.tidyverse.org/reference/read_delim.html

Reply:
updated in https://github.com/umccr/RNAsum/commit/ff43b7291fff1702cf03c55048ee45f10f97c529